k8s/operators/src/pool/main.rs

```diff
@@ -664,6 +677,41 @@ async fn pool_controller(args: ArgMatches) -> anyhow::Result<()> {
             std::process::exit(1);
         }
     }
+    if let Ok(mut v1_alpha1_pools) = list_v1alpha1(&k8s, namespace, PAGINATION_LIMIT).await {
+        if !v1_alpha1_pools.is_empty() {
+            for dsp in v1_alpha1_pools.iter_mut() {
+                if let Some(res_ver) = dsp.resource_version() {
+                    let name = dsp.name_any();
+                    let node = dsp.spec.node();
+                    let disk = dsp.spec.disks();
+                    if let Err(e) = replace_with_v1beta1(
+                        &k8s,
+                        &name,
+                        namespace,
+                        Some(res_ver.clone()),
+                        V1beta1DiskPoolSpec::new(node, disk),
+                    )
+                    .await
+                    {
+                        error!(
+                            "ERROR: failed to create v1beta1 cr for v1alpha1 cr: {:?}",
+                            e
+                        );
+                    } else {
+                        info!("cr creation successful :{}", name);
+                    }
+                } else {
+                    error!(
+                        "ERROR: Cannot get resource version for Pool CR: {}. Skipping CR replace",
+                        dsp.name_any()
+                    );
+                }
+            }
+        }
+    } else {
+        error!("ERROR: while listing for conversion");
```

Comment:
there's a lot of nesting here, is this else for the error? Probably should include the actual error

Reply:
I have removed else arm. It was not required.